idea-plugin/src/main/java/com/palantir/javaformat/intellij/PalantirJavaFormatSettings.java

```diff
@@ -93,15 +94,19 @@ boolean injectedVersionIsOutdated() {
             return true;
         }
 
-        OrderableSlsVersion implementationVersion =
-                OrderableSlsVersion.valueOf(getImplementationVersion().replace(".dirty", ""));
-        OrderableSlsVersion injectedVersion = OrderableSlsVersion.valueOf(formatterVersion.get());
+        Optional<OrderableSlsVersion> implementationVersion =
+                OrderableSlsVersion.safeValueOf(getImplementationVersion().replace(".dirty", ""));
```

Comment:
since `getImplementationVersion()` may now return empty string, we want to avoid `OrderableSlsVersion.valueOf("")` which would throw `SafeIllegalArgumentException` at https://github.com/palantir/sls-version-java/blob/d9d00dd65c98bd6121e837cf1d85938194c1dad0/sls-versions/src/main/java/com/palantir/sls/versions/OrderableSlsVersion.java#L45, so we now handle that case

Reply:
It may be cleaner to return `Optional<String>` to the same end, I think the method is also used here, where a human readable string such as `unknown` may be preferable: https://github.com/palantir/palantir-java-format/blob/9025a224d35424b7a2139fd4aa1ac41c8af75cc7/idea-plugin/src/main/java/com/palantir/javaformat/intellij/PalantirJavaFormatConfigurable.java#L100

Reply:
done